webapp/src/components/Analysis/UtterancesTable.tsx

```diff
@@ -390,6 +403,68 @@ const UtterancesTable: React.FC<Props> = ({
   ];
 
   const searchString = constructSearchString(pipeline);
+
+  const updateProposedAction = (data: UtterancePatch[]) => {
+    const groupByPersistentId: Record<
+      string,
+      UtterancePatch["persistentId"][]
+    > = data.reduce(
+      (
+        accumulated: Record<string, UtterancePatch["persistentId"][]>,
+        { persistentId, dataAction }: UtterancePatch
+      ) => ({
+        ...accumulated,
+        [dataAction]: accumulated[dataAction]
+          ? accumulated[dataAction].concat(persistentId)
+          : [persistentId],
+      }),
+      {}
+    );
+    Object.entries(groupByPersistentId).forEach(
+      ([dataAction, persistentIds]) => {
+        const newValue = dataAction as DataAction;
+        updateDataAction({
+          persistentIds,
+          newValue,
+          ...getUtterancesQueryState,
+        });
+      }
+    );
+  };
+
+  const importProposedActions = (file: File) => {
+    const fileReader = new FileReader();
+    fileReader.onload = ({ target }) => {
+      if (target) {
+        const result = target.result as string;
+        const [header, ...rows] = result.split("\n");
+        if (rows.join()) {
+          const headers: string[] = header.trim().split(",");
+          if (
+            [config.columns.persistent_id, "proposed_action"].every((h) =>
+              headers.includes(h)
+            )
+          ) {
+            const utterancePatch: UtterancePatch[] = rows.map((row) => {
```

Comment:
With the ` as UtterancePatch` bellow, this one is redundant:
```suggestion
            const utterancePatch = rows.map((row) => {
```